scripts/translation/libqa/SyncFileList.php

```diff
@@ -31,43 +31,40 @@ static function load()
         }
 
         $lang = trim( file_get_contents( $file ) );
-        $cache = __DIR__ . "/../../../temp/qaxml.files.$lang";
+        $cacheFilename = __DIR__ . "/../../../temp/qaxml.files.$lang";
 
-        if ( file_exists( $cache ) )
+        if ( file_exists( $cacheFilename ) )
         {
-            $data = file_get_contents( $cache );
+            $data = file_get_contents( $cacheFilename );
             return unserialize( gzdecode( $data ) );
         }
 
+        $sourceDir = 'en';
+        $targetDir = $lang;
+
         require_once __DIR__ . '/../lib/all.php';
 
-        $revcheck = new RevcheckRun( 'en' , $lang );
-        $revdata  = $revcheck->revData;
-        $list = [];
+        $files = new RevcheckFileList( $sourceDir );
+        $ret = [];
 
-        foreach( $revdata->fileDetail as $file )
+        foreach( $files->iterator() as $file )
         {
-            $source = "{$revcheck->sourceDir}/{$file->path}/{$file->name}";
-            $target = "{$revcheck->targetDir}/{$file->path}/{$file->name}";
-
-            if ( ! file_exists( $source ) )
-                continue;
-            if ( ! file_exists( $target ) )
+            if ( ! file_exists( "$targetDir/{$file->file}" ) )
                 continue;
 
             $item = new SyncFileItem();
-            $item->sourceDir = $revcheck->sourceDir;
-            $item->targetDir = $revcheck->targetDir;
-            $item->file = $file->path . '/' . $file->name;
-            $list[] = $item;
+            $item->sourceDir = $sourceDir;
+            $item->targetDir = $targetDir;
+            $item->file = $file->file;
+            $ret[] = $item;
         }
 
-        if ( count( $list ) == 0 )
+        if ( count( $ret ) == 0 )
```

Comment:
```suggestion
        if ( $syncFileItems === [] )
```
When checking for an empty array it's best to use an empty array for clarity. It reads in a more natural way and makes more sense from a type perspective (array comparison instead of int).